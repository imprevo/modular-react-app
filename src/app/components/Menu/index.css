.Menu {
    display: block;
    width: 100%;
    margin: 0;
    padding: 0;
    background: #333;
    font-size: 1rem;
}

.Menu__list {
    width: 100%;
    margin: 0;
    padding: 0;
}

.Menu__item {
    display: inline-block;
    margin: 0;
    padding: 0;
}

.Menu__link {
    display: inline-block;
    padding: .5em 1em;
    color: #fff;
    text-decoration: none;
}
.Menu__link.active,
.Menu__link:hover {
    background: #222;
}