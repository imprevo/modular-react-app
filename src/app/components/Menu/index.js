import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './index.css';

function MenuItem({ children, ...props }) {
    return (
        <li className="Menu__item">
            <NavLink className="Menu__link" activeClassName="active" {...props}>
                {children}
            </NavLink>
        </li>
    )
}

class Menu extends Component {
    render() {
        return (
            <menu className="Menu">
                <ul className="Menu__list">
                    <MenuItem exact to="/">Main</MenuItem>
                    <MenuItem exact to="/about">About</MenuItem>
                    <MenuItem to="/404">404</MenuItem>
                </ul>
            </menu>
        );
    }
}

export { MenuItem };
export default Menu;