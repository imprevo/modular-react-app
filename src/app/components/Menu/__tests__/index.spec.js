import React from 'react';
import { render } from 'react-dom';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';

import Menu, { MenuItem } from '../index';

describe('<Menu /> shallow', () => {
    it('render <Menu />', () => {
        const wrapper = shallow(
            <Menu />
        );
        expect(wrapper.find('.Menu').length).toEqual(1);
        expect(wrapper.find('.Menu__list').length).toEqual(1);
        expect(wrapper.find('.Menu__list').find(MenuItem).length).toEqual(3);
    });

    it('render <MenuItem />', () => {
        const wrapper = shallow(
            <MenuItem expect to={'/test'}>
                some text
            </MenuItem>
        );
        const link = wrapper.find('.Menu__link');

        expect(link.length).toEqual(1);
        expect(link.prop('expect')).toEqual(true);
        expect(link.prop('to')).toEqual('/test');
        expect(link.prop('children')).toEqual('some text');
    });
});

describe('<Menu /> mount', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(
            <MemoryRouter>
                <Menu />
            </MemoryRouter>
        );
    });

    it('render', () => {
        expect(wrapper.find('.Menu').length).toEqual(1);
        expect(wrapper.find(MenuItem).length).toEqual(3);
        expect(wrapper.find(MenuItem).find('.Menu__link').length).toEqual(3);
    });

    it('click on links', () => {
        const links = wrapper.find(MenuItem).find('.Menu__link');

        function checkActivityLinks(active) {
            links.map((link, i) => {
                expect(link.hasClass('active')).toEqual(active === i);
            });
        }

        links.at(0).simulate('click', { button: 0 });
        checkActivityLinks(0);

        links.at(1).simulate('click', { button: 0 });
        checkActivityLinks(1);

        links.at(2).simulate('click', { button: 0 });
        checkActivityLinks(2);
    });
});

describe('<Menu /> snapshot', () => {
    it('renders and matches our snapshot', () => {
        const component = renderer.create(
            <MemoryRouter>
                <Menu />
            </MemoryRouter>
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});