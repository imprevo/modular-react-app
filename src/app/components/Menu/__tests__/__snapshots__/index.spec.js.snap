// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> snapshot renders and matches our snapshot 1`] = `
<menu
  className="Menu"
>
  <ul
    className="Menu__list"
  >
    <li
      className="Menu__item"
    >
      <a
        className="active Menu__link"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        Main
      </a>
    </li>
    <li
      className="Menu__item"
    >
      <a
        className="Menu__link"
        href="/about"
        onClick={[Function]}
        style={undefined}
      >
        About
      </a>
    </li>
    <li
      className="Menu__item"
    >
      <a
        className="Menu__link"
        href="/404"
        onClick={[Function]}
        style={undefined}
      >
        404
      </a>
    </li>
  </ul>
</menu>
`;
